lib/galaxy/datatypes/images.py

```diff
@@ -73,6 +150,40 @@ def handle_dataset_as_image(self, hda: DatasetProtocol) -> str:
             base64_image_data = base64.b64encode(f.read()).decode("utf-8")
         return f"![{name}](data:image/{self.file_ext};base64,{base64_image_data})"
 
+    def set_meta(
+        self, dataset: DatasetProtocol, overwrite: bool = True, metadata_tmp_files_dir: Optional[str] = None, **kwd
+    ) -> None:
+        """
+        Try to populate the metadata of the image using a generic image loading library (pillow), if available.
+
+        If an image has two axes, they are assumed to be ``YX``. If an image has three axes, they are assumed to be ``YXC``.
+        """
+        if PIL is not None:
+            try:
+                with PIL.Image.open(dataset.get_file_name()) as im:
+
+                    # Determine the metadata values that are available without loading the image data
+                    dataset.metadata.width = str(im.size[1])
+                    dataset.metadata.height = str(im.size[0])
```

Comment:
These seem like they should be integers so you can more easily do math ?

Reply:
Makes sense, the reason why I did it as `str` is that when I do it as integers, the validator
```xml
<validator type="dataset_metadata_equal" metadata_name="width" value="32" />
```
fails with the error:
> Metadata value for 'width' must be '32', but it is '32'.

So I was concerned that the metadata cannot be tested properly if it isn't a string.

Am I misusing the validator somehow?

Reply:
Fixed in cdc4507b40e159e838d41f8506b451fa87ea532b